### ISSUE TITLE
create edit and delete not showing up at top of list

### ISSUE THREAD

**Issue by velociwabbit** [marmelab/admin-on-rest#344]
I have been hacking away at Drawer to see if I can integrate it into AOR and it works ok but the create, edit and delete options are not showing up.

I have been trying to figure out how they are instantiated and it is very subtle and hard to track.

Could someone point me in the right direction about where create is declared (I already have the react components that work fine in the non drawer version.)

The nice thing about drawer and   AppNavDrawer is that they are great at controlling the left side precisely. However somewhere in the process the other components are being dropped.

**Comment by fzaninotto** [comment]
Are you looking for the code in `<CrudRoute>`?

https://github.com/marmelab/admin-on-rest/blob/6e82e3a6f47b6a74bf513370de30cb67f751e78a/src/CrudRoute.js#L13-L34

**Comment by velociwabbit** [comment]
​I just stepped through all the crudroute code and it is fine. (all the
booleans hasedit hascreate hasdelete are correct)

Therefore it is likely to be failing because of a filters  warning issue​
that happens when the component mounts somewhere in the depths of this
function:

 markup = ReactReconciler.mountComponent(child, transaction, hostParent,
hostContainerInfo, this._processChildContext(context), debugID);

The warning that is returned is the following and my guess is that the
other components (edit, create etc are ignored)

Warning: Failed prop type: Invalid prop `filters` of type `object` supplied
to `List`, expected a single ReactElement.
    in List_ (created by Connect(List_))
    in Connect(List_) (created by EqupList)
    in EqupList (created by RouterContext)
    in div (created by pure(Component))
    in pure(Component) (created by RouterContext)
    in div (created by BeforeAfterWrapper)
    in BeforeAfterWrapper (created by ClearFix)
    in ClearFix (created by FullWidthSection_)
    in FullWidthSection_ (created by WithWidth)
    in EventListener (created by WithWidth)
    in WithWidth (created by Master_)
    in div (created by Master_)
    in Master_ (created by WithWidth)
    in EventListener (created by WithWidth)
    in WithWidth (created by RouterContext)
    in RouterContext (created by Router)
    in Router (created by Unknown)

The only difference in code is that I am going through appnavbar instead of
admin/layout.  The filters are the same.

**Comment by velociwabbit** [comment]
I figured out the problem.

This is one that is not likely to be repeated very often but it is
something worth thinking about.

As I was combining/ modifying some of the AOR modules there are several
repeated and nonexplicit terms that are used.  For example 'DefaultActions'
is a term used for Actions in  lists, cards and elsewhere.  Some places you
utilize extended names for actions but other places terms like
'defaultState' and 'initialState' are used for the same things.

I spent nearly a week tracking this bug but the bigger issue is the fact
that the call stack for react almost never includes the calling function in
one's own code.  Without specific  and unique names for each variable the
challenge of sorting out the call stack is a  bit of a pain.

If you are interested in the code that uses drawer let me know and I will
publish it.
